_includes/pbc/all/install-features/202304.0/install-the-gift-cards-feature.md

```diff
@@ -0,0 +1,1333 @@
+
+```php
+<?php
+
+namespace Pyz\Zed\Mail;
+se Spryker\Zed\GiftCardMailConnector\Communication\Plugin\Mail\GiftCardUsageMailTypePlugin;
+use Spryker\Zed\Kernel\Container;
+use Spryker\Zed\Mail\Business\Model\Mail\MailTypeCollectionAddInterface;
+use Spryker\Zed\Mail\MailDependencyProvider as SprykerMailDependencyProvider;
+
+class MailDependencyProvider extends SprykerMailDependencyProvider
+{
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return \Spryker\Zed\Kernel\Container
+     */
+    public function provideBusinessLayerDependencies(Container $container)
+    {
+        $container = parent::provideBusinessLayerDependencies($container);
+
+        $container->extend(static::MAIL_TYPE_COLLECTION, function (MailTypeCollectionAddInterface $mailCollection) {
+            $mailCollection->add(new GiftCardUsageMailTypePlugin());
+
+            return $mailCollection;
+        });
+
+        return $container;
+    }
+}
+```
+
+**src/Pyz/Zed/Payment/PaymentDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\Payment;
+
+use Spryker\Zed\GiftCard\Communication\Plugin\GiftCardPaymentMethodFilterPlugin;
+use Spryker\Zed\Kernel\Container;
+use Spryker\Zed\Nopayment\Communication\Plugin\Payment\PriceToPayPaymentMethodFilterPlugin;
+use Spryker\Zed\Payment\PaymentDependencyProvider as SprykerPaymentDependencyProvider;
+
+class PaymentDependencyProvider extends SprykerPaymentDependencyProvider
+{
+    /**
+     * @return \Spryker\Zed\Payment\Dependency\Plugin\Payment\PaymentMethodFilterPluginInterface[]
+     */
+    protected function getPaymentMethodFilterPlugins()
+    {
+        return [
+            new PriceToPayPaymentMethodFilterPlugin(),
+            new GiftCardPaymentMethodFilterPlugin(),
+        ];
+    }
+}
+```
+
+**src/Pyz/Zed/SalesPayment/SalesPaymentDependencyProvider.php**
+
+```php
+<?php
+
+namespace Pyz\Zed\SalesPayment;
+
+use Spryker\Zed\GiftCard\Communication\Plugin\SalesPayment\GiftCardPaymentMapKeyBuilderStrategyPlugin;
+use Spryker\Zed\SalesPayment\SalesPaymentDependencyProvider as SprykerSalesPaymentDependencyProvider;
+
+class SalesPaymentDependencyProvider extends SprykerSalesPaymentDependencyProvider
+{
+    /**
+     * @return array<\Spryker\Zed\SalesPaymentExtension\Dependency\Plugin\PaymentMapKeyBuilderStrategyPluginInterface>
+     */
+    protected function getPaymentMapKeyBuilderStrategyPlugins(): array
+    {
+        return [
+            new GiftCardPaymentMapKeyBuilderStrategyPlugin(),
+        ];
+    }
+}
+```
+
+{% info_block warningBox "Verification" %}
+
+Make sure that:
+* You can activate a gift card using its generated code.
+* You can't activate a gift card the balance of which has been depleted.
+* During the checkout process, payment method selection is skipped in case the gift card covers the grand total.
+* Having made a successful purchase with the help of a gift card, you receive a gift card balance notification e-mail.
```

Comment:
add a verification point that when placing an order with more than one gift card